core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/common/Status.java

```diff
@@ -88,114 +114,29 @@ public boolean isStopped() {
-                DELETE_FAILED,
-                DELETED_ON_PROVIDER_SIDE,
-                DELETE_COMPLETED,
-                STOPPED,
-                STOP_REQUESTED,
-                START_REQUESTED,
-                START_FAILED,
-                STOP_FAILED,
-                WAIT_FOR_SYNC,
-                MAINTENANCE_MODE_ENABLED,
-                AMBIGUOUS,
-                UNREACHABLE,
-                NODE_FAILURE,
-                EXTERNAL_DATABASE_CREATION_IN_PROGRESS,
-                EXTERNAL_DATABASE_CREATION_FAILED,
-                EXTERNAL_DATABASE_DELETION_IN_PROGRESS,
-                EXTERNAL_DATABASE_DELETION_FINISHED,
-                EXTERNAL_DATABASE_DELETION_FAILED,
-                EXTERNAL_DATABASE_START_IN_PROGRESS,
-                EXTERNAL_DATABASE_START_FINISHED,
-                EXTERNAL_DATABASE_START_FAILED,
-                EXTERNAL_DATABASE_STOP_IN_PROGRESS,
-                EXTERNAL_DATABASE_STOP_FINISHED,
-                EXTERNAL_DATABASE_STOP_FAILED,
-                BACKUP_FAILED,
-                BACKUP_FINISHED,
-                RESTORE_FAILED,
-                RESTORE_FINISHED,
-                LOAD_BALANCER_UPDATE_IN_PROGRESS,
-                LOAD_BALANCER_UPDATE_FAILED,
-                LOAD_BALANCER_UPDATE_FINISHED);
+        return STOPPED.equals(this)
+                || STOP_IN_PROGRESS.equals(this)
+                || STOP_REQUESTED.equals(this);
     }
 
     public Status mapToFailedIfInProgress() {
-        if (mappableToFailed().contains(this)) {
-            return convertStatus();
-        }
-        return this;
-    }
-
-    @SuppressWarnings("checkstyle:CyclomaticComplexity")
-    private Status convertStatus() {
-        switch (this) {
-            case REQUESTED:
-            case CREATE_IN_PROGRESS:
-                return CREATE_FAILED;
-            case UPDATE_REQUESTED:
-            case UPDATE_IN_PROGRESS:
-                return UPDATE_FAILED;
-            case RECOVERY_REQUESTED:
-            case RECOVERY_IN_PROGRESS:
-                return RECOVERY_FAILED;
-            case DELETE_IN_PROGRESS:
-            case PRE_DELETE_IN_PROGRESS:
-                return DELETE_FAILED;
-            case START_REQUESTED:
-            case START_IN_PROGRESS:
-                return START_FAILED;
-            case STOP_REQUESTED:
-            case STOP_IN_PROGRESS:
-                return STOP_FAILED;
-            case BACKUP_IN_PROGRESS:
-                return BACKUP_FAILED;
-            case RESTORE_IN_PROGRESS:
-                return RESTORE_FAILED;
-            default:
-                throw new IllegalArgumentException(format("State '%s' is declared to mappable to failed state but not handled.", this));
+        if (isInProgress()) {
+            Status result = IN_PROGRESS_TO_FINAL_STATUS_MAPPING.get(this);
+            if (result == null) {
+                throw new IllegalArgumentException(format("State '%s' is declared to mappable to failed state.", this));
```

Comment:
The original message was more clear to me, or you mean: _`State '%s' is not declared to mappable to failed state.`_ ?

Reply:
Yes, somehow it ended up in this ugly format. I will fix this.